Add Weave Scope to community charts

stefanprodan opened the pull request:
Hello,

This PR adds the [Weave Scope](https://www.weave.works/oss/scope/) open source version to community charts.

Signed-off-by: Stefan Prodan stefan@weave.works

bgower-ibm commented on community/weave-scope/values.yaml:
```diff
@@ -0,0 +1,80 @@
+  basicAuth:
+    enabled: false
+    username: ""
+    password: ""
```
Passwords (or any sensitive data) should really not be entered via values.yaml as it is not secure.  (Any user with permissions to helm can access the helm release manifest and obtain sensitive info).  A better approach is have the user manually pre-create a Secret and then pass in the secret name.

stefanprodan commented:
@bgower-ibm thanks for the review. I've pushed a new chart structure and I've addressed your comments, please let me know if it's ok now. I will push the tgz if no more changes are needed.

bgower-ibm commented:
Have you been testing this on ICP?  I'm still having trouble getting any data from the weave scope UI, which I can get to load the main page now, but it just spins waiting for data from the pods/nodes/etc.

Also, the ClusterRole is excessively permissive.  We need to scope that down to what is required.  I see the public definition of ClusterRole isn't nearly as permissive:  https://cloud.weave.works/k8s/scope.yaml

bgower-ibm commented:
@stefanprodan : Closing as this PR has become stale.  However, we would still really like to get weave-scope in the ICP catalog. The restructure of the sub-charts as done in this PR was probably not necessary.  I'll have my team fork a copy of the public weave-scope chart and get back to you with a diff of our suggested changes that meet the contribution guidelines.
